lib/telephony/pinpoint/sms_sender.rb

```diff
@@ -45,7 +45,11 @@ def send(message:, to:, country_code:, otp: nil)
           )
           finish = Time.zone.now
           response = build_response(pinpoint_response, start: start, finish: finish)
-          return response if response.success?
+          if response.success? ||
+             response.error.is_a?(OptOutError) ||
+             response.error.is_a?(PermanentFailureError)
```

Comment:
Should we consider the SMS opt-in configuration in deciding to fail fast or allow failover?

Reply:
Do you mean `IdentityConfig.store.allow_sms_resubscribe`? Or something else?

Reply:
> Do you mean `IdentityConfig.store.allow_sms_resubscribe`? Or something else?

Yes. In other words, should this logic only apply if we're allowing SMS resubscribe in the environment?

Reply:
My thinking is no, this logic should apply always, opt out means "don't send me SMS" so I think we don't want to failover or retry if we get that response.

We have older opt-out detecting logic that shows a flash error "you have opted out" (that isn't actionable), so this would result in that

Reply:
Makes sense 👍